### ISSUE TITLE
RFD: allow parent images to be removed (with --force)

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#30722]
For historical reasons, the local image store still has code in place to disallow removing "parent" images.

For example;

```bash
$ docker build -t parent -<<EOF
FROM scratch
LABEL foo=bar
EOF

$ docker build -t child -<<EOF
FROM parent:latest
LABEL foo=baz
EOF


$ docker images ls --filter reference=parent
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
parent              latest              17789f62da6c        About an hour ago   0 B

$ docker image rm 17789f62da6c
Error response from daemon: conflict: unable to delete 17789f62da6c (cannot be forced) - image has dependent child images
```

Since the introduction of the content-addressable store, the "child/parent" chain
is only used for the local build cache, but is not required to run an image.

I think we should allow removing the parent image if `--force` is used.

Note that _removing_ the parent image will not cleanup any disk space (because
the _layers_ are still used by the child image), however it does allow users
to clean up images they no longer need.

related discussion https://github.com/docker/docker/issues/25602

/cc @tonistiigi @dmcgowan

**Comment by tonistiigi** [comment]
Would we remove the whole parent chain up to that deleted image then? That may be bit unexpected but if we don't do it then every deletion possibly creates a new dangling image.

**Comment by thaJeztah** [comment]
Good point, I think for any direct descendant of that image, that would be the case yes

**Comment by thaJeztah** [comment]
Looks like this already works if buildkit is enabled, so only an issue if images are built with the classic builder (edited the example steps to use `DOCKER_BUILDKIT=0`
